src/openrct2/interface/Window.cpp

```diff
@@ -2038,50 +2032,48 @@ void window_update_textbox()
     {
         gTextBoxFrameNo = 0;
         rct_window* w = window_find_by_number(gCurrentTextBox.window.classification, gCurrentTextBox.window.number);
-        widget_invalidate(w, gCurrentTextBox.widget_index);
+        widget_invalidate(*w, gCurrentTextBox.widget_index);
         window_event_textinput_call(w, gCurrentTextBox.widget_index, gTextBoxInput);
     }
 }
 
-bool window_is_visible(rct_window* w)
+bool window_is_visible(rct_window& w)
 {
     // w->visibility is used to prevent repeat calculations within an iteration by caching the result
-    if (w == nullptr)
-        return false;
```

Comment:
This is the only exception where a null pointer check inside a function was removed. I've verified that all callers of this function already check for null before this gets called.